src/lifetracker/LifeTracker.java

```diff
@@ -45,4 +57,26 @@ private static void setLogger() throws IOException {
         globalLogger.addHandler(new FileHandler(LOG_FOLDER + LOG_FILE));
 
     }
+
+    @Override
+    public void start(Stage primaryStage) throws Exception {
+        Parent root = FXMLLoader.load(getClass().getResource("/lifetracker/UI/UIDesign.fxml"));
+        Scene scene = new Scene(root);
+        scene.getStylesheets().add(getClass().getResource("/lifetracker/UI/application.css").toExternalForm());
+        primaryStage.setTitle("Life Tracker");
+        primaryStage.setScene(scene);
+        
+        
+        try (Storage fileStorage = new ThreadedFileStorage()) {
```

Comment:
Here, you can do
```Java
//At the top
private Storage fileStorage;

//Remove the try
fileStorage = new ThreadedFileStorage();
//blah blah blah

//Somewhere at the bottom
@Override
public void stop(){
    storageThread.close();
}
```

Then, instead of `System.exit(0)` when you exit in the UI, do `Platform.exit()` instead.